app/app/controllers/application_controller.rb

```diff
@@ -93,12 +93,19 @@ def pilot_ended?
     @pilot_ended.nil? ? current_agency&.pilot_ended : @pilot_ended
   end
 
+  def reset_cbv_session!
+    set_flow_session(nil, nil)
+  end
+
   def set_flow_session(flow_id, type)
-    session[:flow_id] = flow_id
     session[:flow_id] = flow_id
     session[:flow_type] = type
   end
 
+  def flow_class(symbol = nil)
+    (symbol || session[:flow_type]&.to_sym) == :activity ? ActivityFlow : CbvFlow
```

Comment:
nit: `symbol` and `flow_param` are not great names to me since ideally they would convey a bit more meaning. I think `symbol` is actually something like `flow_type`.


```suggestion
  def flow_class(flow_type = session[:flow_type])
    flow_type.to_sym == :activity ? ActivityFlow : CbvFlow
```